tests/node/test_wakunode_filter.nim

```diff
@@ -192,9 +197,8 @@ suite "Waku Filter - End to End":
       server.wakuFilter.subscriptions.subscribedPeerCount() == 1
 
     # And the client node reboots
-    waitFor client.stop()
-    waitFor client.start()
-    discard client.mountFilterClient()
+    await client.stop()
+    await clientClone.start() # Mimic restart by starting the clone
```

Comment:
Why do this through the clone?

Reply:
`client.stop()` does not seem to clean the libp2p switch context completely refuses connections after a `start()`. I investigated but couldn't identify the cause, but using a clone seems like a reasonable workaround to test the same filter functionality. The programmatic starting-stopping-restarting of a node only occurs within test cases and this does not affect app restarts afaict.